### DISCUSSION TITLE
sdm --explore and encrypted rootfs on pi5 w/ different kernels

### DISCUSSION THREAD

**Discussion by tinker2much** [gitbls/sdm#203] [General]
I have a pi5 with an encrypted rootfs (thank you!), I burned some images there using sdm. I wanted to explore one of them, and was told (clearly, thank you again) which line to put into config.txt to force the v8 kernel, in order to enable exploring. 

I did so, and my encrypted rootfs wouldn't work - something missing that I didn't record, sorry - I had to plug that SSD into another machine to remove that line from config.txt to be bootable. 

So, just noting that encrypted rootfs and --explore seem to be incompatible at the moment.

**Comment by gitbls** [comment]
Awww....it's not quite THAT bad. The incompatibility seems to be trying to explore a 32-bit IMG from a 64-bit OS. Doing that gets you the "switch to V8 kernel" message. But, if you switch to the V8 kernel on an encrypted rootfs, the reboot fails spectacularly.

This has nothing to do with sdm, of course. Not sure what the problem is at the moment but will investigate further at some point.

I was able to explore an encrypted 64-bit rootfs from a 64-bit bookworm system:
```
w/ssdy/work# sdm --explore /dev/sdb --encrypted
* Mount device '/dev/sdb'
Enter passphrase for /dev/sdb2: 
mount: /dev/mapper/sdmcrypt mounted on /mnt/sdm.
mount: /dev/sdb1 mounted on /mnt/sdm/boot/firmware.
* Enter Device '/dev/sdb'
...
sdm~# exit
exit
umount: /mnt/sdm/boot/firmware unmounted
umount: /mnt/sdm unmounted
pw/ssdy/work#
```
In the meantime, don't switch to `kernel=kernel8.img` if you're running on an encrypted rootfs.

**Reply by tinker2much** [reply]
I modified the title of this thread to be more specific...

**Reply by unerfurer1s** [reply]
@tinker2much can u provide a complete guide on how u encrypted the root part on rpi5?

**Reply by tinker2much** [reply]
I used bls's sdm-cryptconfig script, see the excellent [docs here](https://github.com/gitbls/sdm/blob/e622451088875c2992a6520d66e58ed8541a99d4/Docs/Disk-Encryption.md)

**Comment by gitbls** [comment]
The next release of sdm will detect this situation:
```
pt/ssd/vanilla/current/armhf# sdm --explore 2024-03-15-raspios-bookworm-armhf-lite.img
* Mount IMG '2024-03-15-raspios-bookworm-armhf-lite.img'
mount: /dev/loop2 mounted on /mnt/sdm.5071.
mount: /dev/loop3 mounted on /mnt/sdm.5071/boot/firmware.
* Enter IMG '2024-03-15-raspios-bookworm-armhf-lite.img'
/bin/bash: error while loading shared libraries: libtinfo.so.6: ELF load command address/offset not page-aligned
?! systemd-nspawn failed on this IMG
   Host system architecture: aarch64 with 16384 byte pages
   IMG binaries: ELF 32-bit LSB executable
?! The rootfs on this host is encrypted; this IMG cannot be processed on this host
umount: /mnt/sdm.5071/boot/firmware unmounted
umount: /mnt/sdm.5071 unmounted
pt/ssd/vanilla/current/armhf# 
```
